src/bin/rbw-agent/actions.rs

```diff
@@ -172,8 +172,36 @@ pub async fn login(
                         )
                         .await?;
                         break;
-                    } else {
-                        return Err(anyhow::anyhow!("TODO"));
+                    } else if providers.contains(
```

Comment:
The way this is implemented means that someone who has an authenticator app enabled as well as a Yubikey will never be able to use the Yubikey to authenticate. 
Is that intended or should we try to go through all methods if pin entry is cancelled or auth fails?

Reply:
Hmm... I hadn't considered that. 
I guess we should honour the order [here](https://bitwarden.com/help/article/setup-two-step-login/#using-multiple-methods)

Reply:
Do you know if that order match the order in `providers` that we get in the Err which is returned? 
That would make it fairly simple to honor that order.

The solution I went for was to simply fall through to the next method if the user cancels the pin entry box, which would maybe mean an extra click if they have both enabled.

Reply:
Looking at the order on that page it definitely does not :)
For me the prompt for authenticator always comes first and then Yubikey after it, which doesn't match the order displayed in the docs.